### PR TITLE
 prov/rxm: progress EQ events in the main thread 

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

**Comment by swelch** [comment]
@a-ilango - Very nice, with this PR I'm seeing upwards of 45% improvement in reducing alltoall times when nearing a 1:1 ratio on ranks:cores.

**Comment by a-ilango** [comment]
> Very nice, with this PR I'm seeing upwards of 45% improvement in reducing alltoall times when nearing a 1:1 ratio on ranks:cores.

Yep. This change gives a very good speed-up to connection scale-up time in an alltoall communication with full subscription. Depending on the node, # of cores, etc., I see improvements from 10x to 70x.

File: prov/rxm/src/rxm_conn.c [hunk 1]
```diff
@@ -599,16 +639,37 @@ int rxm_cmap_process_connreq(struct rxm_cmap *cmap, void *addr,
...
+
+	entry = alloca(RXM_MSG_EQ_ENTRY_SZ);
+	if (!entry) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+			"unable to allocate memory!\n");
+		return -FI_ENOMEM;
+	}
+
+	while (1) {
+		memset(entry, 0, RXM_MSG_EQ_ENTRY_SZ);
```
Comment (shefty) [inline prov/rxm/src/rxm_conn.c]: Is the memset needed inside the loop?

File: prov/rxm/src/rxm_conn.c [hunk 2]
```diff
@@ -663,24 +727,14 @@ static int rxm_cmap_cm_thread_close(struct rxm_cmap *cmap)
...
 	struct rxm_cmap_peer *peer;
 	struct dlist_entry *entry;
 	size_t i;
 
-	rxm_cmap_cm_thread_close(cmap);
-
-	/* cleanup function would be used in manual progress mode */
-	if (cmap->ep->domain->data_progress != FI_PROGRESS_AUTO)
-		rxm_conn_cleanup(cmap->ep);
+	if (cmap->ep->domain->data_progress == FI_PROGRESS_AUTO)
```
Comment (shefty) [inline prov/rxm/src/rxm_conn.c]: Can you move this check inside thread_close and abort if true?

File: prov/rxm/src/rxm_conn.c (hunk 1 above)
Comment (a-ilango) [reply prov/rxm/src/rxm_conn.c]: I think at least the data field of fi_eq_cm_entry needs to be cleared if not the others. This is needed to correctly process FI_CONNECTED event.

File: prov/rxm/src/rxm_conn.c (hunk 2 above)
Comment (a-ilango) [reply prov/rxm/src/rxm_conn.c]: Sure.

File: prov/rxm/src/rxm_conn.c (hunk 1 above)
Comment (a-ilango) [reply prov/rxm/src/rxm_conn.c]: It seems that may not be necessary as we have a check for entry->rd before accessing cm_data. I'll remove this.

**Comment by a-ilango** [comment]
Intel CI failure is for sockets provider. Merging this.

**Pull request merged**